pypxe/http.py

```diff
@@ -55,7 +61,11 @@ def handleRequest(self, connection, addr):
         self.logger.debug('  <--BEGIN MESSAGE-->\n\t{request}\n\t<--END MESSAGE-->'.format(request = repr(request)))
         startline = request.split('\r\n')[0].split(' ')
         method = startline[0]
-        target = startline[1]
+        req_file = startline[1]
+
+        # avoid directory traversal: strip all ../ and make it relative
+        target = os.path.normpath(os.sep + self.netbootDirectory + os.sep + req_file).lstrip(os.sep)
```

Comment:
os.path.join() cannot be used because req_file starts with '/' so all the previous part will be lost.
I added the path traversal filter for platforms that doesn't support os.chroot()